File: web/src/types/review-session.ts
export interface ReviewSession {
    id: number;
    activityId: string;
    mode: 'READ' | 'FIRST_LETTER' | 'CLOZE' | 'QUIZ' | 'SCRAMBLE';
    score: number;
    durationSeconds: number;
    mistakeDetailsJson?: string;
    createdAt: string;
    syncedAt?: string;
}

export interface CreateReviewSessionDto {
    activityId: string; // The UUID of the activity
    mode: 'READ' | 'FIRST_LETTER' | 'CLOZE' | 'QUIZ' | 'SCRAMBLE';
    score: number;
    durationSeconds: number;
    mistakes: string[]; // Array of strings (mistake details)
}

File: mobile/lib/data/models/review_session.dart
import 'package:isar/isar.dart';

part 'review_session.g.dart';

@collection
class ReviewSession {
  Id id = Isar.autoIncrement;

  @Index()
  String? serverId;

  @Index()
  int? activityId; // Local Activity ID reference

  String? mode; // 'QUIZ', 'FILL_BLANK', 'SCRAMBLE'

  int? score;

  int? durationSeconds;

  DateTime? createdAt;

  // JSON String structure: [{"word": "intelligence", "input": "inteligence", "type": "TYPO"}]
  String? mistakeDetailsJson;

  bool isSynced = false;
}
